### DISCUSSION TITLE
onFieldReact中异步的依赖不会被收集 请问是设计如此还是bug

### DISCUSSION THREAD

**Discussion by daweichendoctor** [alibaba/formily#2863] [Q&A 2.x] [answered]
示例如下 https://codesandbox.io/s/xenodochial-wilbur-gqe68n?file=/App.tsx
代码截图如下
![image](https://user-images.githubusercontent.com/30096899/155644389-2d3f2c69-302d-403a-b603-9ced4461e912.png)

此时在input中输入值，reactions不会执行，但是去掉setTimeout后 reactions会执行

**Comment by janryWang** [comment, marked as answer]
先学习下reactive机制吧
